src/main/java/com/example/librarymanagementsystem/model/LibraryResource.java

```diff
@@ -0,0 +1,67 @@
+package com.example.librarymanagementsystem.model;
+
+public class LibraryResource {
+
+    private int resourceId;
+    private String title;
+    private ResourceType resourceType;
+    private String author;
+
+
+    /**
+     * Instantiates a new Library resource.
+     *
+     * @param title  the title
+     * @param author the author
+     */
+    public LibraryResource(String title, String author){
+        this.title = title;
+        this.author = author;
+    }
+
+    /**
+     * Instantiates a new Library resource.
+     *
+     * @param resourceId the resource id
+     * @param title      the title
+     * @param author     the author
+     */
+    public LibraryResource(int resourceId, String title, String author){
+        this.resourceId = resourceId;
+        this.title = title;
+        this.author = author;
+    }
```

Comment:
Instead of defining strict constructors for other parameters, use builders (`Builder Design Pattern`) to make it more flexible, readable and reducing error encounters. Look into that and see how it works.